ociclient/client_test.go

```diff
@@ -218,7 +312,77 @@ func uploadDefaultManifest(ctx context.Context) (*ocispecv1.Manifest, string) {
 			return err
 		}
 	})
-	ref := testenv.Addr + "/test/artifact:v0.0.1"
 	Expect(client.PushManifest(ctx, ref, manifest, ociclient.WithStore(store))).To(Succeed())
-	return manifest, ref
+	return manifest
+}
+
+func compareManifestToTestManifest(ctx context.Context, ref string, manifest *ocispecv1.Manifest) {
+	var configBlob bytes.Buffer
+	Expect(client.Fetch(ctx, ref, manifest.Config, &configBlob)).To(Succeed())
+	Expect(configBlob.String()).To(Equal("test"))
+
+	var layerBlob bytes.Buffer
+	Expect(client.Fetch(ctx, ref, manifest.Layers[0], &layerBlob)).To(Succeed())
+	Expect(layerBlob.String()).To(Equal("test-config"))
+}
+
+func uploadTestIndex(ctx context.Context, indexRef string) *oci.Index {
+	splitted := strings.Split(indexRef, ":")
+	indexRepo := strings.Join(splitted[0:len(splitted)-1], ":")
+	tag := splitted[len(splitted)-1]
+
+	manifest1Ref := fmt.Sprintf("%s-platform-1:%s", indexRepo, tag)
+	manifest2Ref := fmt.Sprintf("%s-platform-2:%s", indexRepo, tag)
+	manifest1 := uploadTestManifest(ctx, manifest1Ref)
+	manifest2 := uploadTestManifest(ctx, manifest2Ref)
+	index := oci.Index{
+		Manifests: []*oci.Manifest{
+			{
+				Descriptor: ocispecv1.Descriptor{
+					Platform: &ocispecv1.Platform{
+						Architecture: "amd64",
+						OS:           "linux",
+					},
+				},
+				Data: manifest1,
+			},
+			{
+				Descriptor: ocispecv1.Descriptor{
+					Platform: &ocispecv1.Platform{
+						Architecture: "amd64",
+						OS:           "windows",
+					},
+				},
+				Data: manifest2,
+			},
+		},
+		Annotations: map[string]string{
+			"test": "test",
+		},
+	}
+
+	tmp, err := oci.NewIndexArtifact(&index)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(client.PushOCIArtifact(ctx, indexRef, tmp)).To(Succeed())
+	return &index
+}
+
+func CompareImageIndices(actualIndex *oci.Index, expectedIndex *oci.Index) {
```

Comment:
docstring missing

Reply:
I've unexported the method, as it is only used for testing